src/js/common/y_axis.js

```diff
@@ -267,7 +267,11 @@ function addNumericalLabels (g, args, axisArgs) {
   var ns = axisArgs.namespace;
   var coords = labelPlacement(args, axisArgs);
   var ticks = args.processed[ns + '_ticks'];
-
+  if(ns === 'x' && ticks.length > 10) {
```

Comment:
This looks like the right thing to change, but the approach doesn't seem quite correct to me. Two things:

1. When the user specifies it explicitly, we should always use their value to determine the number of ticks.
2. When variable, the number of ticks to use should be a function of the available space for the graph, rather than an absolute value.

Reply:
Sure I will make these changes.

Reply:
Sorry @wlach For not updating my work. Will do so today

Reply:
> When the user specifies it explicitly, we should always use their value to determine the number of ticks.

How do I check for this ?
_____
> When variable, the number of ticks to use should be a function of the available space for the graph, rather than an absolute value.

Shall I try to do select ticks on the basis of 10% of width spacing between them ? Or how do you intend on doing this ?

Reply:
It seems as if there are some utility functions in `src/js/utility.js` (e.g. `mg_get_plot_left`) which you could use as a reference for how to get the plot size.

I will leave it to you to come up with an initial heuristic for number of ticks. :) I imagine the current hardcoded value was inspired by the plot sizes used in the metricsgraphics examples -- so using that as a "reference ratio" (# ticks per pixel size of graph) probably makes sense.

Reply:
Cool will incorporate these things